### ISSUE TITLE
Feature: Directional gravity trigger and/or entity

### ISSUE THREAD

**Issue by Klunkyone** [StrataSource/Portal-2-Community-Edition#1701]
### Which component should be improved?

Entity

### Describe your suggestion

I'd love to be able to change the gravity direction of objects (and the player). This would cause the player and other objects to fall in the direction specified, and allow the player to seamlessly move around on a surface that isn't the global floor. Implementing the walking and camera would work similarly to Adhesion gel implementation, with the added bonus of also being able to jump and fall as if the new direction of gravity was normal.

- trigger_gravity_directional

Would simply change the direction of gravity uniformly within the bounds of the trigger.
Has filters and flags to decide what entities this applies to.

Keyvalue to determine direction (Pitch Yaw Roll)
Keyvalue determining scale of gravity (1 is default) (this is similar to trigger_gravity)

- point_gravity_sphere

Ever played Super Mario Galaxy?

![20240729230046_1](https://github.com/user-attachments/assets/64c52809-a0eb-42a1-bca1-f6698fffaf2b)

Would create a sphere of gravity that's downward direction is determined by the objects relation to the point entity.
Basically, you always fall **toward** the origin of the entity.

Keyvalue to determine radius (when leaving radius, gravitational direction returns to default)
Keyvalue to determine scale of gravity (1 is default)
Flag to determine if the force of gravity falls off or is uniform. By default, gravitational influence scales down as you move further away, and is the most intense near the center of the entity. (like real gravity)

I WANNA WALK ON SPHERES! LET ME WALK ON SPHERES!

PS: if either of these get added... the evely exhibition is about to get a whole lot cooler....

**Comment by lenship2** [comment]
Done as of December 23, 2025
